File: TicTacToeJava/src/main/java/Player.java
package TicTacToe;


abstract public class Player {
  private char symbol;
  private String name;

  public Player(char symbol, String name) {
    this.symbol = symbol;
    this.name   = name;
  }

  public char symbol() { return symbol; }

  public String name() { return name; }

  public String toString() { return name; }

  abstract public int getMove(Board board, Player opponent);
}

File: TicTacToe/player.rb
class Player
  attr_reader :mark, :name

  def initialize mark, name
    @mark, @name = mark, name
  end

  def to_s ; name ; end

  # Subclasses should reimplement this method.
  # Given the current board and opposing player,
  # decide which board index you'd like to mark.
  def get_move board, opponent
    gets.chomp.to_i
  end
end